pkg/install/generate.go

```diff
@@ -360,13 +426,15 @@ func GenerateUninstallerJob(
 				"--loglevel",
 				"debug",
 				"--region",
-				cd.Spec.AWS.Region,
-				fmt.Sprintf("kubernetes.io/cluster/%s=owned", cd.Status.InfraID),
-				// Also cleanup anything with the tag for the legacy cluster ID (credentials still using this for example)
-				fmt.Sprintf("openshiftClusterID=%s", cd.Status.ClusterID),
```

Comment:
I don't think we can remove openshiftClusterID here, just did a quick check and registry s3 buckets are still being created only with this tag. It probably wouldn't break deprovision, but it would leak buckets without. Shame as it would be nice to be able to drop that cluster ID from the API.

Reply:
we're not removing it, but only optionally adding it if not present

Reply:
The thinking is that in some deprovision requests you may not have that available anymore. You should at least be able to remove most resources.

Reply:
My mistake I missed it below, this sounds perfect. Should we make ClusterID a pointer to indicate it's optional nature?

Reply:
We can ... it would be a compatible change for the future, but simply leaving it blank does the trick as well.